File: checkout-api/src/main/groovy/ish/oncourse/willow/service/PreferenceApi.java
package ish.oncourse.willow.service;

import ish.oncourse.willow.model.common.CommonError;
import ish.oncourse.willow.model.common.Preferences;

import java.io.InputStream;
import java.io.OutputStream;
import java.util.List;
import java.util.Map;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.MediaType;

@Path("/")
public interface PreferenceApi  {

    @GET
    @Path("/getPreferences")
    @Produces({ "application/json" })
    @CollegeInfo
    Preferences getPreferences();
}

File: checkout/src/js/http/PreferenceApi.ts
import {HttpService} from "../common/services/HttpService";
import {CommonError} from "../model/common/CommonError";
import {Preferences} from "../model/common/Preferences";

export class PreferenceApi {
  constructor(private http: HttpService) {
  }

  getPreferences(): Promise<Preferences> {
    return this.http.GET(`/getPreferences`);
  }
}
